### DISCUSSION TITLE
Creating Diagrams programmatically using Open-BPMN : problem with sequence flows

### DISCUSSION THREAD

**Discussion by zarnejo-exers** [imixs/open-bpmn#142] [Q&A]
Hello!
Greetings!

Thank you very much for this open-bpmn metamodel.
I'm currently trying to use it for my project.
I have already integrated it with the platform I'm using and I can now create models.
However, when I start working on modifying the models, I am unable to properly add sequence flows programmatically (adding events is working well).
When I try to add sequence flows, I encounter an assertion error resulting from an indexoutofbounds exception.

<img width="874" alt="Screenshot 2023-01-12 at 13 27 07" src="https://user-images.githubusercontent.com/3262396/211986375-741ac7e4-41ad-47b8-82e2-ed759a087319.png">

I can properly read a BPMN diagram with sequence flows created using the GUI.
May I know if there's an extra step needed after I add sequence flows and just before I save the model (model.save(out))?

Thank you in advance for taking the time to read my question.
Hoping for a response.

Yours,
Zenith

**Comment by rsoika** [comment]
Hi @zarnejo-exers,  thanks for joining the project. 
I tried to reproduce your error, but without success. I guess there is some more code in your example causing the problem. But of course a IndexOutOfBoundsException should not be thrown by the OpenBPMN Metamodel. 

Here you can see my test example code I used trying to reproduce your problem:

```java
    public void testCreateSequenceFlow() {
        String out = "src/test/resources/output/process-example-8.bpmn";

        String exporter = "demo";
        String version = "1.0.0";
        String targetNameSpace = "http://org.openbpmn";
        BPMNModel model = BPMNModelFactory.createInstance(exporter, version, targetNameSpace);

        try {
            BPMNProcess process = model.openDefaultProcess();
            Event event = process.addEvent("start_1", "Start 1", BPMNTypes.START_EVENT);
            Activity task = process.addTask("task_1", "Task 1", BPMNTypes.TASK);
            process.addSequenceFlow("seq_1", "start_1", "task_1");

            assertNotNull(model);
            assertEquals(1, model.getProcesses().size());
            assertEquals(1, process.getSequenceFlows().size());

        } catch (BPMNModelException e) {
            e.printStackTrace();
            fail();
        }

        model.save(out);
        logger.info("...model created sucessful: " + out);
    }
```

Can you try this out once again? 
Maybe you can post your example code so we can identify the root of the problem...
